Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp

```diff
@@ -1345,9 +1345,9 @@ void GridTrackSizingAlgorithm::accumulateIntrinsicSizesForTrack(GridTrack& track
         // - Items placed at the first implicit line in the masonry axis.
         // - Items that have a specified definite placement in the grid axis.
         // - Items that span all grid axis tracks.
-        if (m_renderGrid->areMasonryRows() || m_renderGrid->areMasonryColumns()) {
+        if (m_renderGrid->isMasonry()) {
             bool skipTrackSizing = true;
-            auto gridAxisDirection = m_renderGrid->areMasonryRows() ? ForColumns : ForRows;
+            auto gridAxisDirection = m_direction;
             auto masonryAxisDirection = m_renderGrid->areMasonryRows() ? ForRows : ForColumns;
             auto span = GridPositionsResolver::resolveGridPositionsFromStyle(*m_renderGrid, *gridItem, gridAxisDirection);
```

Comment:
A few lines below here (1363):

Using `m_direction` still instead of `gridAxisDirection` (same value, but clearer to read).

The first check for !startLine() is unnecessary. If the span is the same as the number of tracks (the second condition), then the start must be at 0 (feel free to add that as an assert).

Reply:
Nice catch.